pkg/mimirtool/config/cortex.go

```diff
@@ -1065,6 +1065,13 @@ var removedConfigPaths = append(gemRemovedConfigPath, []string{
 	"ruler.sharding_strategy",                // -ruler.sharding-strategy
 	"store_gateway.sharding_enabled",         // -store-gateway.sharding-enabled
 	"store_gateway.sharding_strategy",        // -store-gateway.sharding-strategy
+
+	// Removed in 2.1, 2.2 and 2.3
+	"distributor.extend_writes",                          // -distributor.extend-writes
+	"ingester.lifecycler.join_after",                     // -ingester.join-after; is deprecated and will be removed in 2.4, but we can remove it now
```

Comment:
In [Cortex YAML config](https://github.com/cortexproject/cortex/blob/master/docs/configuration/config-file-reference.md) it's at `ingester > lifecycler > ring > join_after`.

```suggestion
	"ingester.lifecycler.ring.join_after",                     // -ingester.join-after; is deprecated and will be removed in 2.4, but we can remove it now
```

Reply:
the join_after in cortex is at the same level as the ring in cortex v1.11 ([code](https://github.com/cortexproject/cortex/blob/v1.11.0/pkg/ring/lifecycler.go#L54)). It looks like it was moved later